emu/src/timer_registers.rs

```diff
@@ -0,0 +1,132 @@
+use crate::bitwise::Bits;
+
+pub enum TimerRegisterAccessControl {
+    Read,
+    Write,
+    ReadWrite,
+}
+
+pub struct Timers {
+    value: u32,
+    access: TimerRegisterAccessControl,
+}
+
+impl Timers {
+    pub fn read(&self) -> u32 {
+        match self.access {
+            TimerRegisterAccessControl::Read | TimerRegisterAccessControl::ReadWrite => self.value,
+            _ => panic!("Requested Timer Register cannot be read."),
+        }
+    }
+
+    pub fn write(&mut self, value: u32) {
+        match self.access {
+            TimerRegisterAccessControl::Write | TimerRegisterAccessControl::ReadWrite => {
+                self.value = value;
+            }
+
+            _ => panic!("Requested Timer Register cannot be written."),
+        }
+    }
+
+    pub fn set_byte(&mut self, byte: u8, value: u8) {
+        match self.access {
+            TimerRegisterAccessControl::Write | TimerRegisterAccessControl::ReadWrite => {
+                self.value.set_byte(byte, value);
+            }
+
+            _ => panic!("Requested Timer Register cannot be written."),
+        }
+    }
+
+    pub const fn with_access_control(access: TimerRegisterAccessControl) -> Self {
+        Self { value: 0, access }
+    }
+}
```

Comment:
I think there's no need to define this struct. In file `io_registers.rs` there's `IORegister` struct

Reply:
Perfect!